test/e2e/proxy_test.go

```diff
@@ -61,10 +64,64 @@ func proxyTestSuite(t *testing.T) {
 	// the deletion of TrustedCAConfigMap, which the former relies on
 	t.Run("Certificate validation", tc.testCertsImport)
 	t.Run("Environment variables validation", tc.testEnvVars)
+
+	t.Run("Request routed through proxy", tc.testProxyRequest)
+	t.Run("Request not proxied", tc.testNoProxyRequest)
+
 	t.Run("Certificate validation removal", tc.testCertsRemoval)
 	t.Run("Environment variables removal validation", tc.testEnvVarRemoval)
 }
 
+// testProxyRequest tests that requests from each Windows node properly go through the proxy
+func (tc *testContext) testProxyRequest(t *testing.T) {
+	clusterProxy, err := tc.client.Config.ConfigV1().Proxies().Get(context.TODO(), "cluster", meta.GetOptions{})
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	for _, node := range gc.allNodes() {
+		t.Run(node.GetName(), func(t *testing.T) {
+			addr, err := controllers.GetAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
+
+			command := "$response = $(Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/openshift/windows-machine-config-operator/master/README.md); $response.Headers['Via']"
+			out, err := tc.runPowerShellSSHJob("get-webrequest-via-header", command, addr)
+			require.NoErrorf(t, err, "unable to retrieve response 'Via' header")
+
+			viaHeader := strings.TrimSpace(out)
```

Comment:
Please make a helper function for this.
such as `func (tc *testContext) getOutboundProxy` or `getOutboundHTTPViaHeaders`
A documented helper will help with code readability.

Reply:
An example of the returned string would be good here.

Reply:
it also stops the duplicated code thats occuring in the noproxy test